### DISCUSSION TITLE
Tool to generate 10-bit HDR YUV in P010 colorspace?

### DISCUSSION THREAD

**Discussion by gregbenz** [google/libultrahdr#64] [Q&A] [answered]
Encoding API-3 (SDR JPG + raw HDR) sounds like it would be very useful for creation of a gain map with full artistic control of both SDR and HDR renditions.

Any suggestions for a tool which may be able to convert some more common HDR format (such as P3 / rec2020 16-bit PQ TIF/PNG or 10-bit AVIF) to the required HDR YUV in P010 colorspace format?

**Comment by DichenZhang1** [comment]
Hi @gregbenz 

I'm happy to answer this question. Unfortunately P010 raw pixel is the only HDR format we currently support but supporting more HDR and SDR formats is a task on our roadmap. We are now planning our Q2 tasks and will be able to share the roadmap very soon.

Thank you!

**Reply by gregbenz** [reply]
Great to hear it, thank you for all the great work you and your team are sharing with the community!

**Reply by ram-mohan** [reply]
update, library now supports raw rgb inputs during encoding. The sample app usage is updated accordingly

**Reply by pmonck** [reply]
Do you know how I could convert an HDR AVIF image into raw rgb inputs? I've been experimenting with converting AVIF to yuv420p10le for input to ultrahdr_app, but my results don't look too good.

**Reply by ram-mohan** [reply, marked as answer]
yuv420p10le and p010 are not identical. Both are 10 bits per component, except that in yuv420p10le, they are the present lsb 10 bits and msb 6 bits are zero and in p010 data is present in msb 10 bits and lsb 6 bits are zero. Also the chroma planes in p010 are interleaved. 
https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/media/java/android/media/MediaCodecInfo.java;l=463

This could be a cause for the problem you are facing.

**Reply by GrousexyHKCN** [reply]
@pmonck i have try ffmpeg, it works, but my input is 16bit .tiff image, ` ffmpeg -i girl.tif -pix_fmt p010le -f rawvideo p010le.raw `, pix_fmt must be p010le. p010be is incorrect.
